### DISCUSSION TITLE
Serverless. Not able to import package in a serverless function

### DISCUSSION THREAD

**Discussion by reflash** [vercel/vercel#4037] [General] [answered]
I'm trying to use mongodb from my serverless function, but i keep getting `mongoose.plugin is not a function` or `mongoose.connect is not a function`. Seems that the problem is it tries to import mongoose, but doesn't find the package.
Could you please advice on that? is there any working sample that does anything similar?

**Comment by reflash** [comment]
here's my project structure to give some context. 

```
.
├── README.md
├── api
│   ├── _helpers
│   │   └── util
│   │       └── index.ts
│   ├── package.json
│   ├── tsconfig.json
│   ├── users
│   │   └── [vkid].ts
│   └── yarn.lock
├── now.json
├── package.json
├── src
│   ├── App.css
│   ├── App.tsx
│   ├── index.css
│   ├── index.tsx
│   ├── react-app-env.d.ts
│   └── serviceWorker.js
├── tsconfig.json
├── tslint.json
└── yarn.lock
```

**Comment by reflash** [comment, marked as answer]
it turned to be a problem specific to mongoose
`import * as mongoose from 'mongoose'` is not working, but `const mongoose = require('mongoose');` does for some reason

**Reply by coetry** [reply]
If your function extension ends in `.js`, then we build it with the latest stable Node.js release which doesn't support the `import` syntax. If you change your extension to `.ts`, we transpile the typescript, which _does_ support that syntax :smile:

**Reply by reflash** [reply]
I am actually using typescript files, this doesn't happen with other dependencies for some reason, so seems more like a mongoose issue

**Comment by bigsee** [comment]
I am having a similar issue when trying to use node-fetch within a serverless function.

`now dev` successfully builds the functions but cannot find the node-fetch module:

> > Building @now/node@latest:api/test.js
> Installing dependencies...
> > Built @now/node@latest:api/test.js [529ms]
> > Building @now/node@latest:api/index.js
> Installing dependencies...
> > Built @now/node@latest:api/index.js [379ms]
> > Building @now/node@latest:api/enquiry.js
> Installing dependencies...
> > Built @now/node@latest:api/enquiry.js [453ms]
> Cannot find module '~/api/node_modules/node-fetch'
> Require stack:
> - /private/var/folders/v9/xxxxxxxxxxxxxxxxxxx/T/zeit-fun-xxxxxxxxxx/api/enquiry.js
> - /private/var/folders/v9/xxxxxxxxxxxxxxxxxxx/T/zeit-fun-xxxxxxxxxx/___now_launcher.js
> - /Users/carltonmcfarlane/Library/Caches/co.zeit.fun/runtimes/nodejs/bootstrap.js
> Did you forget to add it to "dependencies" in `package.json`?

Here is my project structure:

![Screen Shot 2020-04-15 at 13 13 23](https://user-images.githubusercontent.com/11768967/79331091-ed940700-7f1a-11ea-86ca-c87dddf1ea4e.png)

and the `package.json` within `/api`:

```
{
    "name": "api",
    "version": "1.0.0",
    "description": "Serverless API for the project",
    "main": "index.js",
    "scripts": {
        "test": "echo \"Error: no test specified\" && exit 1"
      },
      "author": "bigsee",
      "license": "ISC",
      "dependencies": {
          "node-fetch": "^2.6.0"
     }
}
```

This is blocking me from testing a POST route locally. Any idea what I'm missing?

**Reply by bigsee** [reply]
After a bit more digging it seems that the import path at the top of the serverless function was changed at some point from 

`const fetch = require('node-fetch')`

to

`const fetch = require('~api/node_modules/node-fetch')`

Manually changing back resolves. Perhaps this is VS Code trying to be helpful...
